falcon/response.py

```diff
@@ -136,6 +145,82 @@ def set_stream(self, stream, stream_len):
         self.stream = stream
         self.stream_len = stream_len
 
+    def set_cookie(self, name, value, expires=None, max_age=None,
+                   domain=None, path=None, secure=True, httponly=True):
+        """Set a response cookie.
+
+        Args:
+            name (str):
+                Cookie name
+            value (str):
+                Cookie value
+            expires (datetime): Defines when the cookie should expire.
+                the default is to expire when the browser is closed.
+            max_age (int): The Max-Age attribute defines the lifetime of the
+                cookie, in seconds.  The delta-seconds value is a decimal non-
+                negative integer.  After delta-seconds seconds elapse,
+                the client should discard the cookie.
+            domain (str): The Domain attribute specifies the domain for
+                which the cookie is valid.
+                An explicitly specified domain must always start with a dot.
+                A value of 0 means the cookie should be discarded immediately.
+            path (str): The Path attribute specifies the subset of URLs to
+                which this cookie applies.
+            secure (bool) (default: True): The Secure attribute directs the
+                user agent to use only secure means to contact the origin
+                server whenever it sends back this cookie.
+            httponly (bool) (default: True):
+                The attribute httponly specifies that the cookie
+                is  only transferred in HTTP requests, and is not accessible
+                through JavaScript. This is intended to mitigate some forms
+                of cross-site scripting.
+        Note:
+            Kwargs and their valid values are all
+            specified in http://tools.ietf.org/html/rfc6265
+
+        """
+
+        if self._cookies is None:
+            self._cookies = SimpleCookie()
+
+        self._cookies[name] = value
```

Comment:
Under PY2 this fails:

```python
self._cookies[u'foo'] = 'bar'
```
This can also happen implicitly via:

```python
from __future__ import unicode_literals
```

If the user passes in a `unicode` string, should we try to encode it for them, or just raise an error? At a minimum this should be mentioned in the docstring.

P.S. - It would appear that you can set the value to a `unicode` string, however if it contains any non-ASCII characters, you will still get an error. I didn't test PY3 to see what it does there. We'll have to decide whether to raise an error in this case, or try to convert using UTF-8.